add reference to annotation manifest and fix target source.

robcast opened the pull request:
Adds a reference to manifest of the canvas the annotation targets as `target.source.partOf`. Closes #20
(uses `canvas.options.resource.id` in `AnnotationCreation.js`)

Changes use of `target.id` to `target.source.id`. Closes #25

robcast commented:
Is this ok? Any comments?

mejackreed commented on __tests__/WebAnnotation.test.js:
```diff
@@ -36,21 +35,29 @@ describe('WebAnnotation', () => {
     it('with xywh only', () => {
       subject = createSubject({ svg: null });
-      expect(subject.target()).toBe('canvasId#xywh=xywh');
+      /* expect(subject.target()).toBe('canvasId#xywh=xywh'); */
```
Should we remove this comment?

robcast replied:
Sure, I had left it as a reminder to discuss that WebAnnotation now creates a SpecificResource with a FragmentSelector in all cases. The earlier code would create just a fragment URI if there was only a `xywh` selector (which does not happen anyway).

robcast commented:
I updated the dependency to 3.0.0-rc.5, removed the comment and added nicer `annotation.target.source` handling in the SAS adapter while I was at it :-) `source` (loaded from SAS) will now be a URI if it does not have a manifest link or it will be an object with `id` and `partOf`.
